syngenta_digital_dta/s3/adapter.py

```diff
@@ -238,3 +251,23 @@ def __set_acl(self, public_read):
 
     def __generate_publish_data(self, **kwargs):
         return {'presigned_url': self.create_public_url(**kwargs)}
+
+    def __create_tags(self, **kwargs):
+        tags = kwargs.get('tags', None)
+        result_tags = []
+        try:
+            for key, value in tags.items():
+                result_tags.append({ 'Key' : key, 'Value': value})
+        except AttributeError:
+            pass
+        return result_tags
+
+    def __concat_tags(self, **kwargs):
```

Comment:
and this function should be named like __construct_tags_string() -> the output is the query_string like to supply the Tagging property of the `put()`